ee/tabby-webserver/src/schema/email_service_credential.rs

```diff
@@ -0,0 +1,18 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use juniper::GraphQLObject;
+use tabby_db::EmailServiceCredentialDAO;
+
+#[derive(GraphQLObject)]
+pub struct EmailServiceCredential {
+    pub smtp_username: String,
+    pub smtp_server: String,
+}
+
+#[async_trait]
+pub trait EmailServiceCredentialService: Send + Sync {
+    async fn get_email_service_credential(&self) -> Result<Option<EmailServiceCredential>>;
+    async fn update_email_service_credential(&self, creds: EmailServiceCredentialDAO)
```

Comment:
```suggestion
    async fn update_email_service_credential(&self, smtp_username: &str, smtp_password: &str, smtp_server: &str)
```

Reply:
do not reuse DAO for update input (in future, DAO might add fields like created_at / updated_at / ...)